### ISSUE TITLE
Can't remap the working directory path

### ISSUE THREAD

**Issue by TheGreySquare** [nikita36078/J2ME-Loader#928]
Sorry if it's probably a known issue, I'll just open it up, so anyone else could keep a track of it.

When opening the app for the first time it warns the user that the working directory is placed in the external data storage (/storage/emulated/0/Android/data/ru.playsoftware.j2meloader/) and user can't remap it when going to the settings (tapping on the working directory option doesn't do anything). If I remember correctly, it used to work on pre-Android 11 systems.

Please, consider giving user an option to remap this folder (moving it to an external SD card would be great, too, issue #...), so it would be easier to access and backup it.

Thank you

![Screenshot_20220908-195149_J2ME Loader.png](https://user-images.githubusercontent.com/47947180/189180570-19d5aac9-8327-476e-b644-b8d51ae4ec17.png)

![Screenshot_20220908-193931_J2ME Loader.png](https://user-images.githubusercontent.com/47947180/189180253-88c1fc23-c3e5-4578-a0ed-5f32a5190396.png)

**Comment by xiexindev** [comment]
This is Google's request:
https://developer.android.com/training/data-storage#scoped-storage

To give users more control over their files and to limit file clutter, apps that target Android 10 (API level 29) and higher are given scoped access into external storage, or scoped storage, by default. Such apps have access only to the app-specific directory on external storage, as well as specific types of media that the app has created.

**Comment by DavidAriyan** [comment]
> @xiexindev I know that, but there's still gotta be a way, some workaround to remap the path, after all a lot of apps targeting newer APIs allow you to grant access through DocumentsUI, for example.

Yes many emulators even one that makes changes to game files like rpg maker 2003 emulator (EasyRPG) can set game directory folder in SD card or USB drive. This should be implement here. I'm not a developer but some changes should be made to j2me loader to user convenient. If it can then there's should be a way to backup installed Java app like a button to export import converted Java games so it still can be accessible within the app. There's another alternative solution to this: latest Zarchiver (file manager) can access the app data path by giving it permission. Or use hidden file manager (a loophole in android 13) that's somehow give user access the data & obb folder but it's little complicated (there's a page online about this, Goggle it if you want to know).
